### PR TITLE
Add log event for creating office users

### PR DESCRIPTION
**Pull request by garrettqmartin8**
## Description

In order to meet the needs of the ATO process, we need to store logs of when office users are created, when they were created, and by which user.

## Reviewer Notes

Is this everything we need to comply with the ATO? According to the STIG checklist, this seems to cover us, at least on the account creation front. We'll need to do this work for editing and deleting users as well once those features land.

## Setup

- `make server_run`
- `make client_run`
- Log into the admin app and create a new office user. You should see a new log line like this:
  ```
    2019-09-16T19:47:19.674Z	INFO	adminapi/office_users.go:122	Create Office User	{"git_branch": "gm-ig-log-auditing", "git_commit": "c411bb6da0d45cd995d217ec280f9bb455099c38", "milmove_trace_id": "9bf39b90-66d6-4997-ac80-b58fef656931", "office_user_id": "c323018e-6464-4b3e-b679-c774c6d54263", "responsible_user_id": "bf27c1e7-363a-477f-b68a-e1e9a9742505", "event_type": "created"}
  ```

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Ticket](https://www.pivotaltracker.com/story/show/168439540) for this change

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
Can you experiment with launching this in Experimental and then creating a user and using `ecs-show-logs` to find the event in the logs? That's what I think we need to prove this works.

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj yep, I'll do that.

**Comment by Igarfinkle** [comment]
@garrettqmartin8 @chrisgilmerproj Checked on experimental. The log is there!

**Comment by chrisgilmerproj** [comment]
> @garrettqmartin8 @chrisgilmerproj Checked on experimental. The log is there!

Awesome! Can you do two things for me? Add the log line as a comment here in this PR and/or link to the cloudwatch log. Then I'd love for you to update this README with the example of how to filter for the log:  https://github.com/transcom/mymove/blob/master/cmd/ecs-service-logs/README.md

**Comment by garrettqmartin8** [comment]
@Igarfinkle I'm having trouble finding the log line in experimental using `ecs-service-logs`.

**Comment by chrisgilmerproj** [comment]
Here's how I did it:

```sh
ecs-service-logs show -s app -c app-experimental -e experimental --status STOPPED event_type="created"
```

With this result:

```json
{"level":"info","ts":"2019-09-16T23:43:10.562Z","caller":"adminapi/office_users.go:122","msg":"Create Office User","git_branch":"gm-ig-log-auditing","git_commit":"8ee3b5fca5089c211c1bb14d1571f4791ba092c0","ecs_cluster":"arn:aws:ecs:us-west-2:923914045601:cluster/app-experimental","ecs_task_def_family":"app-experimental","ecs_task_def_revision":"393","milmove_trace_id":"0cbd4167-7dcd-4300-b304-2eed8499695b","office_user_id":"f25f9a15-d4d1-4ba1-a84b-5ae7e33d12c4","responsible_user_id":"1ac2926b-aa8c-4169-9ba7-be6183a96eee","event_type":"created"}
```

The important bit here is that if you are looking at `--status STOPPED` you apparently have to include the cluster name `-c app-experimental` (maybe we need to change that, maybe its ok). I'm also wondering if we want to modify the `event_type` to be something like `create_user` instead of `created` because we don't have a great understanding of what's happening with `created` as it could mean any object type.

If we want to keep `event_type: created` then let's also add something like `table: users`. The problem with separating them is now you have to have multiple filters on your query, which is may be ok or maybe not.

I'd say let's make some decision on this, also update the docs, and then move forward.

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj I've updated the [ecs-service-logs docs](https://github.com/transcom/mymove/blob/master/cmd/ecs-service-logs/README.md) with the example search. I also decided to introduce the convention of a more descriptive `event_type` so we can just use the one field to search on.

I have a question though: why do we need `--status STOPPED` if the default is `ALL`?

**Comment by chrisgilmerproj** [comment]
> I have a question though: why do we need `--status STOPPED` if the default is `ALL`?

I'm not sure why I needed to include that. It could be that `ALL` is broken or just grabbing the newer stuff. Can you experiment with it and debug?

File: cmd/ecs-service-logs/README.md [hunk 1]
```diff
@@ -65,3 +65,12 @@ Filter by url is defined and the number of headers is greater than 14.
 ```shell
 ecs-service-logs show -s app -e experimental url=* "headers>14"
 ```
+
+Search for requests with an event type, such as `create_office_user`
```
Comment (chrisgilmerproj) [inline cmd/ecs-service-logs/README.md]:
```suggestion
Search for requests with an event type, such as `create_office_user`, within the logs for stopped ECS containers:
```

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj hmm, now using `--status STOPPED` in the example you gave doesn't even return the log line. Is it just not able to search yesterday's logs?

**Comment by chrisgilmerproj** [comment]
> @chrisgilmerproj hmm, now using `--status STOPPED` in the example you gave doesn't even return the log line. Is it just not able to search yesterday's logs?

That's curious. Maybe its not searching back past 24 hours by default? The logs are certainly there.

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj haven't had time to dig back in to why the `ecs-service-logs` search behavior is weird, but should we tackle that in a separate ticket? I can add one to the bat team backlog.

**Comment by chrisgilmerproj** [comment]
> @chrisgilmerproj haven't had time to dig back in to why the `ecs-service-logs` search behavior is weird, but should we tackle that in a separate ticket? I can add one to the bat team backlog.

I don't think so. They are co-dependent. If you can log a line but have no way to see that it works then we're kind of up a creek without a paddle. If you want I can be available to look at this with you sometime early next week.

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj that makes sense. Happy to sit down and dig into it with you next week. I'll follow up in Slack.

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj I worked on the `ecs-service-logs` script a bit more but still couldn't manage to get the log event we were searching for even with a very constrained, hard coded time range. However, I was able to find the log event with the following query in `Cloudwatch Insights` with the date range between 9/15 and 9/17.

```
fields @timestamp, msg
| filter msg like /Create Office User/
```

**Comment by chrisgilmerproj** [comment]
@garrettqmartin8 - can you use a different target for that query, specifically `event_type="created"`? What I'm aiming at here is that message text is not reliable but event_type ought to be because that's the point of using `zap.Field`.

**Comment by chrisgilmerproj** [comment]
Btw, the reason the change didn't work is that there's a maximum number of tasks that will be searched and the default is 10, modifiable with the `-t` flag.

**Comment by chrisgilmerproj** [comment]
Here's a minimal implementation of the code I was suggesting: https://github.com/transcom/mymove/pull/2730

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj here's a working, amended Insights query:
```
fields @timestamp, msg, event_type
| filter event_type = "created"
```
Note that the convention for what to name the `event_type` was established after we emitted the original log line in experimental, so going forward, it would be `create_office_user.

I've also added the example you provided in #2730 to the docs, so maybe we can get this merged once that goes in, or vice versa.

**Comment by chrisgilmerproj** [comment]
@garrettqmartin8 - you can save queries in the CloudWatch Insights dashboard. Can you take the query you just created and save it? Then other folks will be able to use it in the future. Would be nice if you could make one for each environment.

**Comment by garrettqmartin8** [comment]
Alrighty, I added the query to each environment dashboard. Merging this and we can continue discussion in #2730. Thanks @chrisgilmerproj!

**Pull request merged**